packages/ui/components/form/selectimproved/components/Options.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useContext, useEffect, useMemo, useState } from "react";
+
+import { useKeyPress } from "@calcom/lib/hooks/useKeyPress";
+
+import { Label } from "../../inputs/Label";
+import Item from "./Item";
+import { SelectContext } from "./SelectProvider";
+import { Option } from "./type";
+
+interface OptionsProps<T extends Option> {
+  list: T[];
+  inputValue: string;
+  isMultiple: boolean;
+  selected: T | T[] | null;
+  searchBoxRef: React.RefObject<HTMLInputElement>;
+}
+
+type FlatternedOption = Option & { current: number; groupedIndex?: number };
```

Comment:
We flattern optional to include the count and also the index of the group it is attached to (If any)